## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 445208`
Commit: `86a429ada70c954ab202d4e3a8013965cd7648f5`

---
### `x86_64-linux`
<details>
  <summary>:x: 8 packages failed to build:</summary>
  <ul>
    <li>beets-unstable (beetsPackages.beets-unstable)</li>
    <li>beets-unstable.dist (beetsPackages.beets-unstable.dist)</li>
    <li>beets-unstable.doc (beetsPackages.beets-unstable.doc)</li>
    <li>beets-unstable.man (beetsPackages.beets-unstable.man)</li>
    <li>beetsPackages.copyartifacts</li>
    <li>beetsPackages.copyartifacts.dist</li>
    <li>beetsPackages.filetote</li>
    <li>beetsPackages.filetote.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 21 packages built:</summary>
  <ul>
    <li>beets (beetsPackages.beets, beetsPackages.beets-stable, python313Packages.beets)</li>
    <li>beets.dist (beetsPackages.beets-stable.dist, beetsPackages.beets.dist, python313Packages.beets.dist)</li>
    <li>beets.doc (beetsPackages.beets-stable.doc, beetsPackages.beets.doc, python313Packages.beets.doc)</li>
    <li>beets.man (beetsPackages.beets-stable.man, beetsPackages.beets.man, python313Packages.beets.man)</li>
    <li>beetsPackages.alternatives</li>
    <li>beetsPackages.alternatives.dist</li>
    <li>beetsPackages.audible</li>
    <li>beetsPackages.audible.dist</li>
    <li>beetsPackages.beets-minimal</li>
    <li>beetsPackages.beets-minimal.dist</li>
    <li>beetsPackages.beets-minimal.doc</li>
    <li>beetsPackages.beets-minimal.man</li>
    <li>python312Packages.beetcamp</li>
    <li>python312Packages.beetcamp.dist</li>
    <li>python312Packages.beets</li>
    <li>python312Packages.beets.dist</li>
    <li>python312Packages.beets.doc</li>
    <li>python312Packages.beets.man</li>
    <li>python313Packages.beetcamp</li>
    <li>python313Packages.beetcamp.dist</li>
    <li>supergee</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>beets-unstable</summary>
<pre>play								play
playlist							playlist
plexupdate							plexupdate
random								random
replace							      &lt;
replaygain							replaygain
rewrite								rewrite
scrub								scrub
smartplaylist							smartplaylist
sonosupdate							sonosupdate
spotify								spotify
subsonicplaylist						subsonicplaylist
subsonicupdate							subsonicupdate
substitute							substitute
the								the
thumbnails							thumbnails
types								types
unimported							unimported
web								web
zero								zero</pre>
</details>
<details>
<summary>beetsPackages.copyartifacts</summary>
<pre>    from beets.library import DefaultTemplateFunctions
E   ImportError: cannot import name &#x27;DefaultTemplateFunctions&#x27; from &#x27;beets.library&#x27; (/nix/store/7dgmyp3li454ps7jm3hly8m64m3gvwxq-beets-2.4.0/lib/python3.13/site-packages/beets/library/__init__.py)
=============================== warnings summary ===============================
../../nix/store/gnk1d1a4lk06r1qyacm38dam96jndh0z-python3.13-confuse-1.7.0/lib/python3.13/site-packages/confuse/util.py:126
  /nix/store/gnk1d1a4lk06r1qyacm38dam96jndh0z-python3.13-confuse-1.7.0/lib/python3.13/site-packages/confuse/util.py:126: DeprecationWarning: &#x27;pkgutil.get_loader&#x27; is deprecated and slated for removal in Python 3.14; use importlib.util.find_spec() instead
    loader = pkgutil.get_loader(name)

tests/helper.py:13
  /build/source/tests/helper.py:13: UserWarning: beets.mediafile is deprecated; use mediafile instead
    from beets import mediafile

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
ERROR tests/test_filename.py
ERROR tests/test_flatdirectory.py
ERROR tests/test_nesteddirectory.py
ERROR tests/test_printignored.py
ERROR tests/test_reimport.py
!!!!!!!!!!!!!!!!!!! Interrupted: 5 errors during collection !!!!!!!!!!!!!!!!!!!!
======================== 2 warnings, 5 errors in 0.18s =========================</pre>
</details>
<details>
<summary>beetsPackages.filetote</summary>
<pre>=========================== short test summary info ============================
ERROR tests/test_audible_m4b_files.py
ERROR tests/test_exclude.py
ERROR tests/test_filename.py
ERROR tests/test_filesize_fixes.py
ERROR tests/test_flatdirectory.py
ERROR tests/test_manipulate_files.py
ERROR tests/test_music_files.py
ERROR tests/test_nesteddirectory.py
ERROR tests/test_pairing.py
ERROR tests/test_patterns.py
ERROR tests/test_printignored.py
ERROR tests/test_reimport.py
ERROR tests/test_rename.py
ERROR tests/test_rename_filetote_fields.py
ERROR tests/test_rename_inline_plugin.py
ERROR tests/test_rename_item_fields.py
ERROR tests/test_rename_paths.py
!!!!!!!!!!!!!!!!!!! Interrupted: 17 errors during collection !!!!!!!!!!!!!!!!!!!
============================== 17 errors in 0.38s ==============================</pre>
</details>
</details>

--- a/pkgs/tools/audio/beets/common.nix
+++ b/pkgs/tools/audio/beets/common.nix
@@ -152,6 +152,7 @@ python3Packages.buildPythonApplication {
       rarfile
       responses
       requests-mock
+      pillow
     ]
     ++ [
       writableTmpDirAsHomeHook

--- a/pkgs/tools/audio/beets/default.nix
+++ b/pkgs/tools/audio/beets/default.nix
@@ -42,23 +42,14 @@ lib.makeExtensible (
 
     beets-minimal = self.beets.override { disableAllPlugins = true; };
 
-    beets-unstable = callPackage ./common.nix {
-      inherit python3Packages;
-      version = "2.3.1";
-      src = fetchFromGitHub {
-        owner = "beetbox";
-        repo = "beets";
-        rev = "d487d675b9115672c484eab8a6729b1f0fd24b68";
-        hash = "sha256-INxL2XDn8kwRYYcZATv/NdLmAtfQvxVDWKB1OYo8dxY=";
-      };
-    };
-
     alternatives = callPackage ./plugins/alternatives.nix { beets = self.beets-minimal; };
     audible = callPackage ./plugins/audible.nix { beets = self.beets-minimal; };
     copyartifacts = callPackage ./plugins/copyartifacts.nix { beets = self.beets-minimal; };
     filetote = callPackage ./plugins/filetote.nix { beets = self.beets-minimal; };
   }
   // lib.optionalAttrs config.allowAliases {
+    beets-unstable = lib.warn "beets-unstable was aliased to beets, since upstream releases are frequent nowadays" self.beets;
+
     extrafiles = throw "extrafiles is unmaintained since 2020 and broken since beets 2.0.0";
   }
 )

--- a/pkgs/tools/audio/beets/plugins/alternatives.nix
+++ b/pkgs/tools/audio/beets/plugins/alternatives.nix
@@ -8,14 +8,14 @@
 
 python3Packages.buildPythonApplication rec {
   pname = "beets-alternatives";
-  version = "0.13.3";
+  version = "0.13.4";
   pyproject = true;
 
   src = fetchFromGitHub {
     repo = "beets-alternatives";
     owner = "geigerzaehler";
     tag = "v${version}";
-    hash = "sha256-j56AzbpZFACXy5KqafE8PCC+zM6pXrxr/rWy9UjZPQg=";
+    hash = "sha256-jGHRoBBXqJq0r/Gbp7gkuaEFPVMGE6cqQRi84AHTXxQ=";
   };
 
   nativeBuildInputs = [
@@ -33,6 +33,7 @@ python3Packages.buildPythonApplication rec {
       pytest-cov-stub
       mock
       pillow
+      tomli
       typeguard
     ]
     ++ [

--- a/pkgs/top-level/aliases.nix
+++ b/pkgs/top-level/aliases.nix
@@ -512,6 +512,7 @@ mapAliases {
   BeatSaberModManager = beatsabermodmanager; # Added 2024-06-12
   beam_nox = throw "beam_nox has been removed in favor of beam_minimal or beamMinimalPackages"; # Added 2025-04-01
   beatsabermodmanager = throw "'beatsabermodmanager' has been removed due to lack of upstream maintainenance. Consider using 'bs-manager' instead"; # Added 2025-03-18
+  inherit (beetsPackages) beets-unstable;
   betterbird = throw "betterbird has been removed as there were insufficient maintainer resources to keep up with security updates"; # Added 2024-10-25
   betterbird-unwrapped = throw "betterbird has been removed as there were insufficient maintainer resources to keep up with security updates"; # Added 2024-10-25
   bfc = throw "bfc has been removed, as it does not build with supported LLVM versions"; # Added 2025-08-10

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -2392,7 +2392,7 @@ with pkgs;
   };
 
   beetsPackages = lib.recurseIntoAttrs (callPackage ../tools/audio/beets { });
-  inherit (beetsPackages) beets beets-unstable;
+  inherit (beetsPackages) beets;
 
   binlore = callPackage ../development/tools/analysis/binlore { };
 

